wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -93,12 +94,12 @@ getHistory mCWalId mAccountId mAddrId = do
     allAccIds <- getWalletAccountIds cWalId
 
     let getAccAddrs = map cwamId <$> concatMapM (getAccountAddrsOrThrow Ever) accIds
-        noAccFiltering = accIds == allAccIds
+        noAccFiltering = HS.fromList accIds == HS.fromList allAccIds
```

Comment:
Want to guess what the worst-case complexity of this is?

Reply:
1. Documentation for `fromList` says _O(n*min(W, n))_: https://hackage.haskell.org/package/unordered-containers-0.2.8.0/docs/Data-HashSet.html#v:fromList
2. I have no idea what `W` is -- it could be an implementation constant, like tree fanout factor.
3. Considering the worst case, let's assume `W ~ n`.
4. With all of the above assumptions, the worst-case complexity seems to be `O(n^2)`, on the total number of accounts.

Reply:
Inspection of `fromList`:
- `HashSet . List.foldl' (\ m k -> H.insert k () m) H.empty)` -- a series of inserts proportional on:
    - the length of the input
    - complexity of `Data.HashMap.Lazy.insert`

Reply:
`Data.HashMap.Lazy.insert` is a classic tree insert, and is documented to have `O(log n)`, but that likely is the average case, with balanced distribution across the tree.  The worst case should be `O(n)`, where all the entries are lined up in a linear chain, tree-structure-wise.

Given this assessment of `Data.HashMap.Lazy.insert`, and the above implementation of `Data.HashSet.fromList`, my final assessment for the `fromList` complexity is `O(n^2)`.